packages/conversation/src/models/v1/app-message-message/app-message-message.ts

```diff
@@ -10,12 +10,120 @@ import { ContactInfoMessage } from '../contact-info-message';
 
 /** The content of the message */
 export type AppMessageMessage =
-  CardMessage
-  | CarouselMessage
-  | ChoiceMessage
-  | LocationMessage
-  | MediaMessage
-  | TemplateMessage
-  | TextMessage
-  | ListMessage
-  | ContactInfoMessage;
+  AppMessageCardMessage
+  | AppMessageCarouselMessage
+  | AppMessageChoiceMessage
+  | AppMessageLocationMessage
+  | AppMessageMediaMessage
+  | AppMessageTemplateMessage
+  | AppMessageTextMessage
+  | AppMessageListMessage
+  | AppMessageContactInfoMessage;
+
+interface AppMessageCardMessage extends CardMessage {
```

Comment:
If these edge cases are currently handled manually: a good candidate to improve auto generation from OAS file (and dedicated and typescript generator)to avoid possible human error by missing some exclusion